### ISSUE TITLE
ArrayType causes org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column found (Types#ARRAY), but expecting (Types#OTHER)

### ISSUE THREAD

**Issue by Novalaina** [vladmihalcea/hypersistence-utils#697]
**Describe the bug**
When I upgrade from 

`hypersistence-utils-hibernate-62:3.7.1`
to
`hypersistence-utils-hibernate-63:3.7.1`

The following code: 

 ```
@Type(
      value = ListArrayType::class,
      parameters = [Parameter(
          value = "house",
          name = ListArrayType.SQL_ARRAY_TYPE,
      )]
  )
  @Column(columnDefinition = "house[]")
  val houses: List<House>,
```

Throws this error:

`org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [houses] in table [dok_card]; found [_house (Types#ARRAY)], but expecting [house[] (Types#OTHER)]`

My hibernate version is `6.4.1.Final`.

**To Reproduce**

Follow the instructions here: https://vladmihalcea.com/postgresql-array-java-list/ with the latest version of hibernate and hypersistence-utils.

**Expected behavior**
No error is thrown when using a postgresql array of enums with a java list.

**Comment by vladmihalcea** [comment]
The validation will no longer work since without `Types.OTHER`, it will not work at all. So, either you disable the schema validation or you switch to the Hibernate native ARRAY Type.

If [HHH-17588](https://hibernate.atlassian.net/browse/HHH-17588) will be fixed, then we can switch back to `Types.ARRAY`. In the meantime, there's no other way.

**Comment by flozano** [comment]
Just as a reference: I got affected by the same when upgrading from 3.7.0 to 3.7.2 with Hibernate 6.4 (which I think wasn't fully supported before :)

**Comment by vladmihalcea** [comment]
This issue is a side effect of #698. However, without it, you wouldn't be able to use the ArrayTypes from this library at all with Hibernate 6.4.

Check out my previous message for more details about it.

**Comment by flozano** [comment]
Yes... I've tried moving to native array, but then when I trigger an update my enums try to get mapped to positional enum.
`[ERROR: column "roles" is of type company_role[] but expression is of type smallint[]`

**Comment by vladmihalcea** [comment]
You should open a Hibernate issue then. In the long run, it's better to have the native Type support everything this library has been supporting.

**Comment by GuiF007** [comment]
@vladmihalcea when you say "native Type support", it's implementation of UserType<String[]> for example ? (for a array of string)

Do you confirm your fix make it works better with Hibernate 6.4+ and only validation fail ?

**Comment by GuiF007** [comment]
Sure ! Just wanted to know there is no known regression than the validation. 
We will check to update to native support in any case.

**Comment by vladmihalcea** [comment]
The only known regressions are the ones for which there are Open issues on GitHub. There might be unknown regressions, of course.

**Comment by kth13** [comment]
Hi @vladmihalcea , I am facing the similar issue The code we have

**Comment by vladmihalcea** [comment]
@kth13 Remove the `@Type` annotation and see if it works.

**Comment by kth13** [comment]
@vladmihalcea It worked, thanks a lot.

**Comment by vladmihalcea** [comment]
Great. The native Hibernate Type should be fine in many situations.

**Comment by Javasick** [comment]
How to implement array of SQL enum in 6.4? 
```
    @Type(
        value = EnumArrayType.class,
        parameters = @Parameter(
            name = AbstractArrayType.SQL_ARRAY_TYPE,
            value = "status"
        )
    )
    @Column(name = "statuses", columnDefinition = "status[]")
    private Status[] statuses;
```
Native Type support cause the following error on saving:
`column "statuses" is of type status[] but expression is of type smallint[]`

**Comment by vladmihalcea** [comment]
@Javasick There is a [`PostgreSQLEnumArrayTypeTest`](https://github.com/vladmihalcea/hypersistence-utils/blob/master/hypersistence-utils-hibernate-63/src/test/java/io/hypersistence/utils/hibernate/type/array/PostgreSQLEnumArrayTypeTest.java#L133) that shows you how to use the `EnumArrayType` with Hibernate 6.4.

**Comment by Javasick** [comment]
That approach causing schema validation error
`Schema-validation: wrong column type encountered in column [statuses] in table [mapping]; found [_status (Types#ARRAY)], but expecting [status[] (Types#OTHER)]`

**Comment by hahosseini** [comment]
> @Javasick There is a [`PostgreSQLEnumArrayTypeTest`](https://github.com/vladmihalcea/hypersistence-utils/blob/master/hypersistence-utils-hibernate-63/src/test/java/io/hypersistence/utils/hibernate/type/array/PostgreSQLEnumArrayTypeTest.java#L133) that shows you how to use the `EnumArrayType` with Hibernate 6.4.

how about hibernate 6.5 ? This solution does not work here

**Comment by vladmihalcea** [comment]
It works just fine with Hibernate 6.5. The 63 module is built with Hibernate 6.5.

https://github.com/vladmihalcea/hypersistence-utils/blob/master/hypersistence-utils-hibernate-63/pom.xml#L207

**Comment by hahosseini** [comment]
> It works just fine with Hibernate 6.5. The 63 module is built with Hibernate 6.5.
> 
> https://github.com/vladmihalcea/hypersistence-utils/blob/master/hypersistence-utils-hibernate-63/pom.xml#L207

for this definition : 
```
@Type(value = EnumArrayType.class,
 parameters = @Parameter(name = AbstractArrayType.SQL_ARRAY_TYPE, value = “tag_enum”))
@Column(nullable = false, columnDefinition = "tag_enum[]")
Tag[] tags;
```
 hibernate 6.5 and 63 module (version 3.8.2) I'm  getting : 
`Schema-validation: wrong column type encountered in column [tags] in table  ; found [__tag_enum (Types#ARRAY)], but expecting [tag_enum[] (Types#OTHER)]`

**Comment by vladmihalcea** [comment]
@hahosseini Well, you said that the `PostgreSQLEnumArrayTypeTest` solution doesn't work, and it does work. 

Now, it's obvious that you are getting that `wrong column type` warning since this issue is still open, right? 

The only way to get rid of that warning is if you fix [HHH-17588](https://hibernate.atlassian.net/browse/HHH-17588) in Hibernate. Be the change you want to see in this world!

**Comment by soufianefhiyil** [comment]
i still have the same issue => Schema-validation: wrong column type encountered in column [valeurs] in table [champs_composant_specifiques]; found [_text (Types#ARRAY)], but expecting [text[] (Types#OTHER)]

  @Type(value = StringArrayType.class)
  @Column(name = VALEURS, columnDefinition = "text[]")
  private String[] valeurs;
  
  CREATE TABLE champs_composant_specifiques (
  id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
  name VARCHAR NOT NULL,
  valeurs TEXT[],
  composant_id INTEGER REFERENCES composant (id) ON DELETE CASCADE
);

**Comment by snigdhasjg** [comment]
Since `HHH-17588` is now closed, what's the next step on this?

I'm stuck upgrading hibernate from `6.2.42` to `6.6.36` (latest)
For my use case, I'm using `ListArrayType.class` to convert it to `List<String>`
```java
    @Type(ListArrayType.class)
    List<String> tags;
```

**Comment by vladmihalcea** [comment]
Fixed in `hypersistence-utils-hibernate-71` since the [HHH-17588](https://hibernate.atlassian.net/browse/HHH-17588) was fixed only in Hibernate ORM 7.1.

**Comment by vladmihalcea** [comment]
The fix had a regression and had to be rolled back. It was causing #821
